### PR TITLE
Open frame conn automaton

### PR DESCRIPTION
**Pull request by ragansa**
Encapsulates frame management/serialization/de-serialization inside ConnectionAutomaton

### CODE REVIEW

File: src/ConnectionAutomaton.h [hunk 1]
```diff
@@ -186,7 +186,7 @@ class ConnectionAutomaton final
     return streamsFactory_;
   }
 
-  FrameSerializer& frameSerializer() const override;
+  FrameSerializer& frameSerializer() const ;//override;
```
Comment (lexs) [inline src/ConnectionAutomaton.h]: what is this?

File: src/automata/ChannelRequester.cpp [hunk 2]
```diff
@@ -130,15 +130,15 @@ void ChannelRequester::endStream(StreamCompletionSignal signal) {
   ConsumerBase::endStream(signal);
 }
 
-void ChannelRequester::onNextFrame(Frame_PAYLOAD&& frame) {
+void ChannelRequester::handlePayload(Payload&& payload, bool complete, bool flagsNext) {
```
Comment (lexs) [inline src/automata/ChannelRequester.cpp]: bool arguments is kinda horrible, why exactly are we no longer passing the frame? if we really don't want to pass the frame could we create some sort of Flags type that allows easier querying? `flags.has(FrameFlags::COMPLETE)`?

File: src/automata/ChannelRequester.cpp [hunk 3]
```diff
@@ -147,14 +147,14 @@ void ChannelRequester::onNextFrame(Frame_PAYLOAD&& frame) {
...
-  processPayload(std::move(frame.payload_), frame.header_.flagsNext());
+  processPayload(std::move(payload), flagsNext);
 
   if (end) {
     closeStream(StreamCompletionSignal::COMPLETE);
   }
 }
 
-void ChannelRequester::onNextFrame(Frame_ERROR&& frame) {
+void ChannelRequester::handleError(std::string errorPayload) {
```
Comment (lexs) [inline src/automata/ChannelRequester.cpp]: this should pass an `folly::exception_wrapper`

File: src/automata/ChannelResponder.cpp [hunk 4]
```diff
@@ -91,18 +91,20 @@ void ChannelResponder::processInitialFrame(Frame_REQUEST_CHANNEL&& frame) {
       true);
 }
 
+/*
```
Comment (lexs) [inline src/automata/ChannelResponder.cpp]: what's going on here? why not remove this if we no longer need it?

File: src/automata/RequestResponseRequester.cpp [hunk 5]
```diff
@@ -71,7 +71,7 @@ void RequestResponseRequester::endStream(StreamCompletionSignal signal) {
   }
 }
 
-void RequestResponseRequester::onNextFrame(Frame_ERROR&& frame) {
+void RequestResponseRequester::handleError(std::string errorPayload) {
```
Comment (lexs) [inline src/automata/RequestResponseRequester.cpp]: same here

File: src/ConnectionAutomaton.cpp [hunk 6]
```diff
@@ -568,8 +561,75 @@ void ConnectionAutomaton::onConnectionFrame(
...
+                               frame_payload.header_.flagsNext());
+      break;
+    }
+    case FrameType::ERROR: {
+      Frame_ERROR frame_error;
+      if (!deserializeFrameOrError(frame_error,
+                                   std::move(serializedFrame))) {
+        return;
+      }
+      automaton->handleError(frame_error.payload_.moveDataToString());
```
Comment (ragansa) [inline src/ConnectionAutomaton.cpp]: @lexs 
I've updated the later methods to use folly::exception_wrapper, but I'm not clear on how to construct one that will capture the error frame's payload.

Comment (lexs) [inline src/ConnectionAutomaton.cpp]: You'll need to give it an actual exception, `std::runtime_error("the_string")` is probably fine for now.

Comment (ragansa) [inline src/ConnectionAutomaton.cpp]: That did it!  Thanks!

**Comment by lehecka** [comment]
The summary for this PR is rather short and not descriptive.
Scott, can you update why this change and what is the intent for the code changes?

File: src/ConnectionAutomaton.h (hunk 1 above)
Comment (lehecka) [inline src/ConnectionAutomaton.h]: I believe we can remove the method now as the intent for this PR is to not use Frames and FrameSerializer outside of ConnectionAutomaton.

**Comment by ragansa** [comment]
I've made a huge mess of this, I'm going to try to get a clean PR together

**Pull request closed without merging**